cmd/gcp-controller-manager/app/gke_approver.go

```diff
@@ -124,14 +127,23 @@ func (a *gkeApprover) handle(csr *capi.CertificateSigningRequest) error {
 			csrApprovalLatency.WithLabelValues("sar_error", r.authFlowLabel).Observe(time.Since(start).Seconds())
 			return err
 		}
-		if approved {
-			glog.Infof("validator %q: SubjectAccessReview approved for CSR %q", r.name, csr.Name)
-			csrApprovalStatus.WithLabelValues("approve", r.authFlowLabel).Inc()
-			csrApprovalLatency.WithLabelValues("approve", r.authFlowLabel).Observe(time.Since(start).Seconds())
-			return a.updateCSR(csr, true, r.approveMsg)
-		} else {
+		if !approved {
 			glog.Warningf("validator %q: SubjectAccessReview denied for CSR %q", r.name, csr.Name)
+			continue
+		}
+		glog.Infof("validator %q: SubjectAccessReview approved for CSR %q", r.name, csr.Name)
+		if r.preApproveHook != nil {
+			if err := r.preApproveHook(a.opts, csr, x509cr, r.authFlowLabel, a.client); err != nil {
+				glog.Warningf("validator %q: preApproveHook failed for CSR %q", r.name, csr.Name)
```

Comment:
I think the error itself only gets logged at v(4), which I'm guessing we don't often have enabled.  Should we log it here?  (Worst case, we double log!)

Reply:
+1 we should add err to the message

Reply:
Done.